src/linear_algebra/quadratic_form.lean

```diff
@@ -788,3 +789,189 @@ theorem exists_orthogonal_basis [hK : invertible (2 : K)]
 let ⟨v, hv₁, hv₂, _⟩ := exists_orthogonal_basis' hB₁ hB₂ in ⟨v, hv₁, hv₂⟩
 
 end bilin_form
+
+/-- Nonzero elements in a field are invertible. This is a defition since it requires a parameter. -/
+def field.invertible {K : Type*} [field K] {z : K} (hz : z ≠ 0) : invertible z :=
+{ inv_of := z⁻¹,
+  inv_of_mul_self := inv_mul_cancel hz,
+  mul_inv_of_self := mul_inv_cancel hz }
+
+namespace quadratic_form
+
+open_locale big_operators
+
+open finset bilin_form
+
+variables {M₁ : Type*} [add_comm_group M₁] [module R M₁]
+variables {ι : Type*} [fintype ι] {v : ι → M}
+
+/-- A quadratic form composed with a linear equiv is isometric to itself. -/
+def isometry_of_comp_linear_equiv (Q : quadratic_form R M) (f : M₁ ≃ₗ[R] M) :
+  Q.isometry (Q.comp (f : M₁ →ₗ[R] M)) :=
+{ map_app' :=
+  begin
+    intro,
+    simp only [comp_apply, linear_equiv.coe_coe, linear_equiv.to_fun_eq_coe,
+               linear_equiv.apply_symm_apply, f.apply_symm_apply],
+  end, .. f.symm }
+
+/-- Given a quadratic form `Q` and a basis, `of_is_basis` is the basis representation of `Q`. -/
+noncomputable def of_is_basis (Q : quadratic_form R M)
+  (hv₁ : is_basis R v) : quadratic_form R (ι → R) := Q.comp hv₁.equiv_fun.symm
+
+@[simp]
+lemma isometry_of_is_basis_apply (Q : quadratic_form R M) (hv₁ : is_basis R v)
+  (w : ι → R) : Q.of_is_basis hv₁ w = Q (∑ i : ι, w i • v i) :=
+by { rw ← hv₁.equiv_fun_symm_apply, refl }
+
+/-- A quadratic form is isometric to its bases representations. -/
+noncomputable def isometry_of_is_basis (Q : quadratic_form R M) (hv₁ : is_basis R v) :
+  isometry Q (Q.of_is_basis hv₁) :=
+isometry_of_comp_linear_equiv Q hv₁.equiv_fun.symm
+
+lemma isometry_of_is_Ortho_apply [invertible (2 : R₁)]
+  (Q : quadratic_form R₁ M) (hv₁ : is_basis R₁ v)
+  (hv₂ : (associated Q).is_Ortho v) (w : ι → R₁) :
+  Q.of_is_basis hv₁ w = ∑ i : ι, associated Q (v i) (v i) * w i * w i :=
+begin
+  rw [isometry_of_is_basis_apply, ← @associated_eq_self_apply R₁, sum_left],
+  refine sum_congr rfl (λ j hj, _),
+  rw [sum_right, sum_eq_single j],
+  { rw [smul_left, smul_right], ring },
+  { intros i _ hij,
+    rw [smul_left, smul_right,
+        show (associated_hom R₁) Q (v j) (v i) = 0, by exact hv₂ i j hij,
+        mul_zero, mul_zero] },
+  { intro hnj, exact false.elim (hnj hj) }
+end
+
+/-- The weighted sum of squared with respect some weight. `weighted_sum_squares` is the
+quadratic form version of this. -/
+def weighted_sum_squares' (w : ι → R) : (ι → R) → R :=
```

Comment:
You could consider generalizing to 
```suggestion
def weighted_sum_squares' [algebra S R] (w : ι → S) : (ι → R) → R :=
```
and using `•` instead of `*` to apply the weights.

Reply:
Since I'm no longer using `weighted_sum_squares'`, I feel we should just remove it